cmd/server/assets/login/login.html

```diff
@@ -114,6 +120,49 @@
             }
           });
       });
+
+      $pinForm.on('submit', function(event) {
+        event.preventDefault();
+
+        // Disable the submit button so we only attempt once.
+        $submitPin.prop('disabled', true);
+
+        // Ask user for the SMS verification code.
+        let cred = firebase.auth.PhoneAuthProvider.credential(verId, $pin.val());
+        let multiFactorAssertion = firebase.auth.PhoneMultiFactorGenerator.assertion(cred);
+        // Complete sign-in.
+        resolver.resolveSignIn(multiFactorAssertion)
+          .catch(function(err) {
+            clearExistingFlash();
+            flash(err.message, "danger");
+            $submit.prop('disabled', false);
+          });
+      });
+
+      $pinClose.on('click', function(event) {
+        $submit.prop('disabled', false);
+        $loginDiv.show();
+        $pinDiv.hide();
+      });
+
+      $resendPin.on('click', function(event) {
```

Comment:
Do you need to prevent default here?

Reply:
it's a link with no href, not a submit button?